contributors/design-proposals/storage/csi-migration.md

```diff
@@ -237,15 +245,35 @@ with the in-tree plugin, the VolumeAttachment object becomes orphaned.
 ### In-line Volumes
 
 In-line controller calls are a special case because there is no PV. In this case
-we will translate the volume source and copy it to the field
-VolumeAttachment.Spec.Source.VolumeAttachmentSource.InlineVolumeSource. The
-VolumeAttachment name must be made with the CSI Translated version of the
+we will translate the in-line volume source into a CSIPersistentVolumeSource 
+and copy it to the field `VolumeAttachment.Spec.Source.VolumeAttachmentSource.InlineCSIVolumeSource`.
+The VolumeAttachment name must be made with the CSI Translated version of the
 VolumeSource in order for it to be discoverable by Detach and WaitForAttach
 (described in more detail below).
 
-The CSI Attacher will have to be modified to also check this location for a
-source as well as checking for the `pvName`. Only one of the two may be
-specified.
+The CSI Attacher will have to be modified to also check for `InlineCSIVolumeSource`
+besides the `PersistentVolumeName`. Only one of the two may be specified. If `PersistentVolumeName` 
+is empty and `InlineCSIVolumeSource` is set, the CSI Attacher will not look for 
+an associated PV in it's PV informer cache as it implies the inline volume scenario 
+(where no PVs are created).
+
+The CSI Attacher will have access to most of the data it requires for handling in-line 
+volumes attachment (through the CSI plugins) from `InlineCSIVolumeSource`. However 
+a few fields require special handling due to the absence of a PV where such fields
+are assigned desired values. Specifically, reasonable `MountOptions` and 
```

Comment:
I guess the question is how are inline volumes handled currently? We want the MountOptions and AccessModes for each plugin to match the default (or chosen?) values/behavior that would have been done without migration.

It is worth calling out in the design that is the philosophy used when choosing the per-plugin defaults.

Reply:
In-line volumes don't have `MountOptions`. Volume plugin defaults are always used.